src/UnitySampleProject/Assets/Samples/web3.unity SDK Web3Auth/2.5.1/Web3.Unity Web3Auth Samples/Scripts/Web3AuthLoginProvider.cs

```diff
@@ -47,55 +62,58 @@ public void SetRememberMe(bool rememberMe)
     protected override async void Initialize()
     {
         base.Initialize();
-
-        //Always first add listeners.
         providerAndButtonPairs.ForEach(p =>
             p.Button.onClick.AddListener(delegate { LoginWithWeb3Auth(p.Provider); }));
-
-#if UNITY_WEBGL && !UNITY_EDITOR
-        Uri uri = new Uri(Application.absoluteURL);
-
-        // make sure this load isn't redirected from Web3Auth a login
-        if (!string.IsNullOrEmpty(uri.Fragment))
-        {
-            useProvider = false;
-
-            await TryLogin();
-           
-        }
-#else
+        #if !UNITY_EDITOR && UNITY_WEBGL
+        Web3AuthWebGLConnected += Web3AuthSet;
+        var projectSettings = ProjectConfigUtilities.Load();
```

Comment:
You should use the IChainConfig that is registered in DI instead, as a user can provide a different config than the one returned by ProjectConfigUtilities.Load().